### ISSUE TITLE
NonApiType: signaling for use of non-API-types on private methods does not make sense

### ISSUE THREAD

**Issue by cobratbq** [google/error-prone#4254]
[NonApiType](<https://errorprone.info/bugpattern/NonApiType>) warns for concrete types used as return types for (static) private methods. To verify this for private methods does not make sense, given that the class is the unit of abstraction. You cannot ignore the chosen concrete type if it is subsequently used as part of internal state.

Note that you can voluntarily choose to ignore the concrete type if its use is so minimal that the concrete type does not in any way reach an implementation boundary or design limitation. However, given that private methods can only be accessed in the internals of classes, the class itself needs to be sound unit and hiding types would obscure possible problems.

**Comment by octylFractal** [comment]
I agree, I think it's noisy to require `@SuppressWarnings("NonApiType")` for a private method that wants to take an `ArrayList` so that it can call `ensureCapacity` on it.

**Comment by cpovirk** [comment]
I agree in principle, so I did a search. There were a surprising (to me) number of hits in our depot, and of the ten in the random sample that I looked at, none needed the more specific collection type.

I still think it could be nice to do something here, presumably based on whether any of the additional members of the subtype are actually used. I'm just disappointed that the solution doesn't look to be uncontroversially "just ignore private members."
